devportal/devportal-domain-service/ballerina/applicationImpl.bal

```diff
@@ -86,3 +86,72 @@ function deleteApplication(string appId, string organization) returns string|err
     error?|string status = deleteApplicationDAO(appId,organization);
     return status;
 }
+
+function generateAPIKey(APIKeyGenerateRequest payload, string appId, string keyType, string user, string org) returns APIKey|error {
+    string?|Application|error application = getApplicationById(appId, org);
+    if application !is Application {
+        return error("Invalid Application. Application with application id:" + appId + " not found");
+    } else {
+        boolean userAllowed = checkUserAccessAllowedForApplication(application, user);
+        if userAllowed {
+            int validityPeriod = 0;
+            int? payloadValPeriod = payload.validityPeriod;
+            if payloadValPeriod is int {
+                validityPeriod = payloadValPeriod;
+            } 
+            record {} addProperties = {};
+            record {}? payloadAddProperties = payload.additionalProperties;
+            if payloadAddProperties is record {} {
+                addProperties = payloadAddProperties;
+            }
+
+            // retrieve subscribed APIs
+            string?|SubscriptionList|error subscriptions =  getSubscriptions(null,appId,null,0,0,org);
```

Comment:
I suppose we have formatting issues in here. shall we do a `bal format` for these changes?
```suggestion
            string?|SubscriptionList|error subscriptions =  getSubscriptions(null, appId, null, 0, 0, org);
```

Reply:
fixed